ComunityMembers/RepositoryTemplate/AbstractRepository.cs

```diff
@@ -0,0 +1,60 @@
+﻿using CommunityMembersModels.Contracts;
+using CommunityRepository.DatabaseContext;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CommunityRepository
+{
+    public abstract class AbstractRepository<T> where T : IModel
+    {
+        public FullDbContext _dataContext { get; set; }
+
+        public AbstractRepository(FullDbContext fullContext)
+        {
+            _dataContext = fullContext;
+        }
+        public void Delete(int Id)
+        {
+            Task<T> item = GetAsync(Id);
+            if(item.Result != null)
```

Comment:
This it does not works as you expected. The `GetAsync` method which is executed asynchronously needs to be finished when the null check it is done, otherwise the result will be null even if the records exists in db.

Reply:
`        public async void Delete(int Id)
        {
            T item = await GetAsync(Id);
            if(item != null)`

Is it the correct way?

Reply:
should be: 
```
public async Task DeleteAsync(int Id)
{
}
```